lib/puppet/provider/user/directoryservice.rb

```diff
@@ -37,6 +37,12 @@
 ## Class Methods ##
 ##               ##
 
+  # This method holds the authentication authority key needed for managing
+  # a valid and loggable macOS 11 user with ShadowHash
+  def self.authentication_authority
+    ";ShadowHash;HASHLIST:<SALTED-SHA512-PBKDF2,SRP-RFC5054-4096-SHA512-PBKDF2>"
+  end
```

Comment:
Does it make sense for this to be a public class method that is used inside this class instance methods? Wouldn't be better to have it as a private method? or just a plain constant since it is just a string?

Reply:
I'd turn it into a constant and name it `SHA512_PBKDF2_AUTHENTICATION_AUTHORITY` or something that also includes the hash type.